### PR TITLE
Custom activation functions for Encoder

### LINKED ISSUE #1039: Allow other activation functions for Encoder variance

### PR DESCRIPTION
**Pull request by PierreBoyeau**
FIxes #1039

### CODE REVIEW

File: scvi/nn/_base_components.py [hunk 1]
```diff
@@ -282,7 +289,7 @@ def forward(self, x: torch.Tensor, *cat_list: int):
         # Parameters for latent distribution
         q = self.encoder(x, *cat_list)
         q_m = self.mean_encoder(q)
-        q_v = torch.exp(self.var_encoder(q)) + self.var_eps
+        q_v = self.var_activation(self.var_encoder(q))
```
Comment (adamgayoso) [inline scvi/nn/_base_components.py:292]:
```suggestion
        q_v = self.var_activation(self.var_encoder(q)) + self.var_eps
```

File: scvi/nn/_base_components.py [hunk 2]
```diff
@@ -235,6 +235,7 @@ def __init__(
         dropout_rate: float = 0.1,
         distribution: str = "normal",
         var_eps: float = 1e-4,
+        var_activation: Optional[Callable] = None,
```
Comment (adamgayoso) [inline scvi/nn/_base_components.py:238]:
```suggestion
        var_activation: Optional[Callable] = torch.exp,
```

File: tests/models/test_models.py [hunk 3]
```diff
@@ -34,6 +35,9 @@ def test_scvi(save_path):
     model = SCVI(adata, n_latent=n_latent)
     model.train(1, check_val_every_n_epoch=1, train_size=0.5)
 
+    model = SCVI(adata, n_latent=n_latent, var_activation=Softplus())
```
Comment (adamgayoso) [inline tests/models/test_models.py:38]: the SCVI class would need this option which would get passed to the encoder

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/module/_vae.py [hunk 4]
```diff
@@ -96,6 +96,7 @@ def __init__(
         use_batch_norm: Literal["encoder", "decoder", "none", "both"] = "both",
         use_layer_norm: Literal["encoder", "decoder", "none", "both"] = "none",
         use_observed_lib_size: bool = True,
+        var_activation: Optional[Callable] = torch.exp,
```
Comment (adamgayoso) [inline scvi/module/_vae.py:99]: need an entry in the docstring for this

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/module/_vae.py [hunk 5]
```diff
@@ -96,6 +98,7 @@ def __init__(
...
```
Comment (adamgayoso) [inline scvi/module/_vae.py:101]: Actually let's make this None, and use torch.exp if None. Be sure to add in docstring, If `None`, defaults to :func:`~torch.exp`.

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/module/_vae.py (hunk 5 above)
Comment (PierreBoyeau) [reply scvi/module/_vae.py]: Done!

**Review by adamgayoso — APPROVED** [review]

**Pull request merged**